modules/core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/BaseDocument.java

```diff
@@ -691,17 +691,13 @@ protected void setValueBlob(T state, Blob blob, String xpath) throws PropertyExc
     }
 
     protected void setPropertyBlobData(String xpath, String string) {
+        Blob blob = StringUtils.isNotEmpty(string) ? Blobs.createBlob(string) : null;
+        DocumentBlobManager blobManager = Framework.getService(DocumentBlobManager.class);
         String key;
-        if (string == null) {
-            key = null;
-        } else {
-            Blob blob = Blobs.createBlob(string);
-            DocumentBlobManager blobManager = Framework.getService(DocumentBlobManager.class);
-            try {
-                key = blobManager.writeBlob(blob, this, xpath);
-            } catch (IOException e) {
-                throw new PropertyException("Cannot write binary for doc: " + getUUID(), e);
-            }
+        try {
+            key = blobManager.writeBlob(blob, this, xpath);
+        } catch (IOException e) {
```

Comment:
Not sure if I'm seeing this correctly but, would it make sense to just declare `key` and call `setPropertyValue` inside the _try_ block?

Reply:
It's preferable to have try/catch blocks cover just what's needed and nothing more.